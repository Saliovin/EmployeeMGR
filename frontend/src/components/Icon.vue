<script setup lang="ts">
import BootstrapIcons from "bootstrap-icons/bootstrap-icons.svg";
import type { PropType } from "vue";

enum IconSizes {
    "sm",
    "md",
    "lg"
}

defineProps({
    icon: {
        type: String,
        required: true,
    },
    size: {
        type: String as PropType<keyof typeof IconSizes>,
        default: IconSizes.md,
    },
});
</script>

<template>
    <svg class="icon" :class="size">
        <g transform-origin="center">
            <use v-bind:xlink:href="`${BootstrapIcons}#${icon}`" />
        </g>
    </svg>
</template>

<style scoped>
.icon {
    width: 1em;
    height: 1em;
}

.sm {
    font-size: 0.75rem;
}

.md {
    font-size: 1rem;
}

.lg {
    font-size: 1.5rem;
}
</style>