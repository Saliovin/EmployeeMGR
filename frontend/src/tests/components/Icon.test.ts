import Icon from '@/components/Icon.vue'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

describe('icon', () => {
    it('base', () => {
        const card = mount(Icon, {
            props: {
                icon: 'x',
                size: 'lg'
            }
        })
        expect(card.element).toMatchSnapshot()
    })
})