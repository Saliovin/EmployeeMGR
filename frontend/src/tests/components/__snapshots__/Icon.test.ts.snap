// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`icon > base 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <svg
        class="icon lg"
        data-v-26446d8e=""
      >
        <g
          data-v-26446d8e=""
          transform-origin="center"
        >
          <use
            data-v-26446d8e=""
            xlink:href="/node_modules/bootstrap-icons/bootstrap-icons.svg#x"
          />
        </g>
      </svg>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {
        "VTU_COMPONENT": {},
      },
      "da": null,
      "data": {},
      "devtoolsRawSetupState": {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": {},
          "_value": {},
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "_depsLength": 5,
        "_dirtyLevel": 0,
        "_runnings": 0,
        "_shouldSchedule": false,
        "_trackId": 1,
        "active": true,
        "allowRecurse": true,
        "deps": [
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "scheduler": [Function],
        "trigger": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {
        "VTU_COMPONENT": {},
      },
      "refs": {
        "VTU_COMPONENT": {},
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "_depsLength": 5,
            "_dirtyLevel": 0,
            "_runnings": 0,
            "_shouldSchedule": false,
            "_trackId": 1,
            "active": true,
            "allowRecurse": true,
            "deps": [
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
            "trigger": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {
        "VTU_COMPONENT": {},
      },
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "icon": "x",
            "size": "lg",
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "BootstrapIcons": "/node_modules/bootstrap-icons/bootstrap-icons.svg",
            "IconSizes": {
              "0": "sm",
              "1": "md",
              "2": "lg",
              "lg": 2,
              "md": 1,
              "sm": 0,
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "_depsLength": 2,
            "_dirtyLevel": 0,
            "_runnings": 0,
            "_shouldSchedule": false,
            "_trackId": 1,
            "active": true,
            "allowRecurse": true,
            "deps": [
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
            "trigger": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {},
          "exposed": {},
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "icon": "x",
            "size": "lg",
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "icon": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
              "size": {
                "0": false,
                "1": true,
                "default": 1,
                "type": [Function],
              },
            },
            [
              "size",
            ],
          ],
          "provides": {},
          "proxy": {
            "icon": "x",
            "size": "lg",
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "_depsLength": 2,
                "_dirtyLevel": 0,
                "_runnings": 0,
                "_shouldSchedule": false,
                "_trackId": 1,
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
                "trigger": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {
            "BootstrapIcons": "/node_modules/bootstrap-icons/bootstrap-icons.svg",
            "IconSizes": {
              "0": "sm",
              "1": "md",
              "2": "lg",
              "lg": 2,
              "md": 1,
              "sm": 0,
            },
          },
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": [
                      "xlink:href",
                    ],
                    "el": <use
                      data-v-26446d8e=""
                      xlink:href="/node_modules/bootstrap-icons/bootstrap-icons.svg#x"
                    />,
                    "key": null,
                    "patchFlag": 8,
                    "props": {
                      "xlink:href": "/node_modules/bootstrap-icons/bootstrap-icons.svg#x",
                    },
                    "ref": null,
                    "scopeId": "data-v-26446d8e",
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "targetStart": null,
                    "transition": null,
                    "type": "use",
                  },
                ],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <g
                  data-v-26446d8e=""
                  transform-origin="center"
                >
                  <use
                    data-v-26446d8e=""
                    xlink:href="/node_modules/bootstrap-icons/bootstrap-icons.svg#x"
                  />
                </g>,
                "key": null,
                "patchFlag": 0,
                "props": {
                  "transform-origin": "center",
                },
                "ref": null,
                "scopeId": "data-v-26446d8e",
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "targetStart": null,
                "transition": null,
                "type": "g",
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": [
                  "xlink:href",
                ],
                "el": <use
                  data-v-26446d8e=""
                  xlink:href="/node_modules/bootstrap-icons/bootstrap-icons.svg#x"
                />,
                "key": null,
                "patchFlag": 8,
                "props": {
                  "xlink:href": "/node_modules/bootstrap-icons/bootstrap-icons.svg#x",
                },
                "ref": null,
                "scopeId": "data-v-26446d8e",
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "targetStart": null,
                "transition": null,
                "type": "use",
              },
            ],
            "dynamicProps": null,
            "el": <svg
              class="icon lg"
              data-v-26446d8e=""
            >
              <g
                data-v-26446d8e=""
                transform-origin="center"
              >
                <use
                  data-v-26446d8e=""
                  xlink:href="/node_modules/bootstrap-icons/bootstrap-icons.svg#x"
                />
              </g>
            </svg>,
            "key": null,
            "patchFlag": 2,
            "props": {
              "class": "icon lg",
            },
            "ref": null,
            "scopeId": "data-v-26446d8e",
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "targetStart": null,
            "transition": null,
            "type": "svg",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/home/cubone/Sprout-Exam/frontend/src/components/Icon.vue",
            "__name": "Icon",
            "__scopeId": "data-v-26446d8e",
            "components": {},
            "props": {
              "icon": {
                "required": true,
                "type": [Function],
              },
              "size": {
                "default": 1,
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <svg
          class="icon lg"
          data-v-26446d8e=""
        >
          <g
            data-v-26446d8e=""
            transform-origin="center"
          >
            <use
              data-v-26446d8e=""
              xlink:href="/node_modules/bootstrap-icons/bootstrap-icons.svg#x"
            />
          </g>
        </svg>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "icon": "x",
          "ref": "VTU_COMPONENT",
          "size": "lg",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "targetStart": null,
        "transition": null,
        "type": {
          "__file": "/home/cubone/Sprout-Exam/frontend/src/components/Icon.vue",
          "__name": "Icon",
          "__scopeId": "data-v-26446d8e",
          "components": {},
          "props": {
            "icon": {
              "required": true,
              "type": [Function],
            },
            "size": {
              "default": 1,
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <svg
          class="icon lg"
          data-v-26446d8e=""
        >
          <g
            data-v-26446d8e=""
            transform-origin="center"
          >
            <use
              data-v-26446d8e=""
              xlink:href="/node_modules/bootstrap-icons/bootstrap-icons.svg#x"
            />
          </g>
        </svg>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "targetStart": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.4.38",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {},
  "isDisabled": [Function],
  "rootVM": {
    "VTU_COMPONENT": {},
  },
  "wrapperElement": <svg
    class="icon lg"
    data-v-26446d8e=""
  >
    <g
      data-v-26446d8e=""
      transform-origin="center"
    >
      <use
        data-v-26446d8e=""
        xlink:href="/node_modules/bootstrap-icons/bootstrap-icons.svg#x"
      />
    </g>
  </svg>,
}
`;
